src/VocabularyManager.Core/Interfaces/IVocabularyStoreManager.cs

```diff
@@ -0,0 +1,14 @@
+﻿using VocabularyManager.Core.Entities;
+
+namespace VocabularyManager.Core.Interfaces
+{
+    public interface IVocabularyStoreManager
+    {
+        Task<int> AddWord(Word word, int wordListId);
+        Task AddWords(IEnumerable<Word> words, int wordListId);
```

Comment:
why no return value but by AddWord you have return value?
this could return `ImmutableList<int>`

https://stackoverflow.com/questions/30165810/why-use-immutablelist-over-readonlycollection

or `AddWord` should not return anything. Consistancy it's important :)